folium/plugins/heat_map.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+"""
+Heat map
+--------
+
+Create a HeatMap layer
+"""
+import json
+from jinja2 import Template
+
+from folium.element import JavascriptLink, Figure
+from folium.map import TileLayer
+
+class HeatMap(TileLayer):
+    def __init__(self, data, name=None,
+                 min_opacity = .5, max_zoom=18, max_val=1.,
+                 radius=25, blur=15, gradient=None,
+                 overlay = True):
+        """Create a Heatmap layer
+
+        Parameters
+        ----------
+        data : list of points of the form [lat, lng, weight]
+            The point you want to plot.
```

Comment:
Let's support 3D numpy as well.  We can listify them internally.

Question:  In your test example you pass a 2D array and things work.  I guess that we should raise an error instead, no?